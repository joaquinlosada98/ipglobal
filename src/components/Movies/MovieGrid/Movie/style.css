.Movie{
  align-content: center;
  border: 1px solid #fcfcfc;        
  border-radius: 6px;
  margin: 0 10px 10px 0;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  background-color:  var(--bg-primary);
}

.Movie img{
  height: 260px;
  cursor: pointer;
}

.Movie p {
  margin: 10px 0;
  text-align: center;
  color: #fcfcfc;
}
