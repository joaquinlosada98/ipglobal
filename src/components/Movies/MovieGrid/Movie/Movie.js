import React from 'react';
import { useHistory } from 'react-router-dom';
import './style.css';

const Movie = ({id, title, poster_path, release_date}) => {

    const history = useHistory();
    
    const handleClick = () => {
        history.push(`/${id}`)
    }
    
    return (
        <div className="Movie animate__animated animate__fadeIn">
            <img src={"https://www.themoviedb.org/t/p/w220_and_h330_face"+ poster_path} 
                alt={title}
                onClick={handleClick}
            />
            <p>{title}</p>
            <p>Release: {release_date}</p>
        </div>
    )
}

export default Movie