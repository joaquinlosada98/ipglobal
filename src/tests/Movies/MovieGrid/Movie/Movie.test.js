import React from 'react';
import { shallow } from 'enzyme';
import Movie from '../../../../components/Movies/MovieGrid/Movie/Movie'

describe('Pruebas en <Movie/>', () => {
    const id = 123;
    const title = "Un titulo";
    const poster_path = "https://www.themoviedb.org/t/p/w220_and_h330_face/6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg";
    const release_date = "un fecha";


    const wrapper = shallow(<Movie
        id
        title 
        poster_path
        release_date
    />)

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })


})