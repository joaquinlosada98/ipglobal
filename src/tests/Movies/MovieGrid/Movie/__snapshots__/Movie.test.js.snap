// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Movie/> debe de mostrar el componente correctamente 1`] = `
<div
  className="Movie animate__animated animate__fadeIn"
>
  <img
    alt={true}
    onClick={[Function]}
    src="https://www.themoviedb.org/t/p/w220_and_h330_facetrue"
  />
  <p>
    <Component />
  </p>
  <p>
    Release: 
    <Component />
  </p>
</div>
`;
